src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -251,21 +251,19 @@ func handleRegistrationFlow(
 		}
 	}
 
-	// Check if a registration flow has been completed successfully
-	for _, flow := range cfg.Derived.Registration.Flows {
-		if checkFlowsEqual(flow, authtypes.Flow{sessions[sessionID]}) {
-			return completeRegistration(req.Context(), accountDB, deviceDB,
-				r.Username, r.Password, r.InitialDisplayName)
+	// Check if the user's registration flow has been completed successfully
+	if !checkFlowCompleted(authtypes.Flow{sessions[sessionID]}, cfg.Derived.Registration.Flows) {
```

Comment:
it feels odd that we are constructing a `Flow` here and then immediately unpacking it in `checkFlowCompleted`. Why not just pass in the `[]LoginType` ?

Reply:
My inner OCD wanted to only accept Flow's into checkFlowCompleted, but yes I can change it :)